networking/p2p/src/main/java/tech/pegasys/artemis/networking/p2p/jvmlibp2p/rpc/RpcMessageHandler.java

```diff
@@ -130,20 +131,15 @@ public ResponderHandler(Connection connection) {
     protected void channelRead0(ChannelHandlerContext ctx, ByteBuf byteBuf) {
       STDOUT.log(Level.DEBUG, "Responder received " + byteBuf.array().length + " bytes.");
       Bytes bytes = Bytes.wrapByteBuf(byteBuf);
-      TRequest request = RpcCodec.decode(bytes, method.getRequestType());
+      TRequest request = rpcCodec.decodeRequest(bytes, method.getRequestType());
```

Comment:
> If steps (1), (2), or (3) fail due to invalid, malformed, or inconsistent data, the responder MUST respond in error. Clients tracking peer reputation MAY record such failures, as well as unexpected events, e.g. early stream resets.

If decoding fails, should we respond with an error?

Reply:
Yes.  Logged BC-88.